kep/NNN-preemption-toleration/README.md

```diff
@@ -0,0 +1,125 @@
+
+However, it is hard to monitor when the first epoch is finished outside of the job. Thus, a minimum running time guarantee is useful practically.  For example, cluster administrator would introduce several priority classes with the same priority value for introducing several minimum running time guarantee categories like below:
+
+```yaml
+                low: 80000  # no minimum running time guarantee
+low-10min-guarantee: 80000  # 10 min. minimum running time guarantee
+low-30min-guarantee: 80000  # 30 min. minimum running time guarantee
+```
+
+### Lower priority value but not-being-preempted priority
+
+As described in the previous section, the job needs to preserve its state outside of the pod at preemption for making it resumable.  However, when a job uses some proprietary tools that are not modifiable and do not support suspend/resume feature,  users would like to avoid being preempted. 
+
+To achieve this, `PriorityClass` with a high priority value would be enough. But just giving high priority value causes preemption on any pods with lower priority values.  Sometimes, users don't want to do it.  In such a case, the job's priority is essentially low (i.e. the job is expected to run only when the cluster has a vacancy).  However, once scheduled, users don't want the pod to be preempted because the pod is difficult to support export/import its state.
+
+On the other hand, from the cluster administrative view, unconditional non-preempted pods are not desirable because there are system-critical, node-critical pods in the cluster. So, the cluster administrator would like to set the minimum priority value for such a priority class.  For example, assume there are three priority classes: system-critical, high and low.  Then, the cluster administrator would like to introduce `low-non-preempted` priority class that can not be preempted by high but can be preempted by system-critical.
+
+```yaml
+  system-critical: 10000
+             high:  9000
+              low:  8000
+low-non-preempted:  8000 # with minimumPriority=10000
+                        # (i.e. high can't preempt this, but system-critical can preempt this)
+```
+
+## Design Details
+
+### Preemption Toleration API
+
+Preemption toleration policy is defined in this struct type.
+
+```go
+// Please note that this does NOT have ObjectMeta 
+// because this will be embedded in annotation in PriorityClass as unnamed object.
+type PreemptionToleration struct {
+	metav1.TypeMeta `json:",inline"`
+
+  // MinimumPreemptablePriority specifies the minimum priority value that can preempt this priority class.
+  MinimumPreemptablePriority *int32 `json:"minimumPreemptablePriority,omitempty"`
+
+  // MinimumRunningDuration specified the minimum running duration of pods that can be preempted.
+  MinimumRunningDuration *time.Duration `json:"minimumRunningDuration,omitempty"`
```

Comment:
Are these two conditions OR'e or AND'ed?

Reply:
I thought AND'ed.